app/models/collection.rb

```diff
@@ -167,8 +167,12 @@ def title
   scope :by_title, -> { order(:title) }
   scope :for_blurb, -> { includes(:parent, :moderators, :children, :collection_preference, owners: [:user]).with_attached_icon }
 
-  def cleanup_url
-    self.header_image_url = Addressable::URI.heuristic_parse(self.header_image_url) if self.header_image_url
+  def cleanup_url   
+    self.header_image_url = begin 
+      Addressable::URI.heuristic_parse(self.header_image_url) if self.header_image_url
+    rescue Addressable::URI::InvalidURIError
+      self.header_image_url
```

Comment:
This is a good start, however the URL could now pass the other validation because it's not guaranteed to cover everything that `Addressable::URI.heuristic_parse` errors on. That can be solved by using the `url_format` validator for the URL instead of the current validation, which checks the heuristic parse as well as a regex. This is what we do for example in `prompt.rb`.